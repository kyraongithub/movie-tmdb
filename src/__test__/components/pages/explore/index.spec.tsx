import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { MemoryRouter } from 'react-router-dom';
import ExplorePage from '../../../../components/pages/explore';

describe('Explore', () => {
  it('renders Explore component', () => {
    const queryClient = new QueryClient();
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <MemoryRouter initialEntries={['/explore?q=test']}>
          <ExplorePage />
        </MemoryRouter>
      </QueryClientProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
