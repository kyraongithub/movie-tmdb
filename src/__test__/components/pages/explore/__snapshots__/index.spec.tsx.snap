// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Explore > renders Explore component 1`] = `
<div>
  <div
    class="min-h-[100vh] bg-gray-900 text-white"
  >
    <section
      class="pt-20 pb-12 px-4 sm:px-6 lg:px-8 bg-gradient-to-b from-gray-800 to-gray-900"
    >
      <div
        class="max-w-3xl mx-auto text-center"
      >
        <h1
          class="text-3xl sm:text-4xl font-bold mb-6"
        >
          Explore Movies
        </h1>
        <form
          class="flex gap-2 w-full"
        >
          <div
            class="relative flex-1"
          >
            <input
              class="w-full pl-10 pr-4 py-3 bg-gray-800 border border-gray-700 rounded-lg focus:ring-2 focus:ring-primary focus:border-transparent"
              placeholder="Search for movies..."
              type="text"
              value=""
            />
          </div>
          <button
            class="cursor-pointer px-6 py-3 bg-primary hover:bg-primary-dark transition-colors"
            type="submit"
          >
            Search
          </button>
        </form>
      </div>
    </section>
    <section
      class="py-8 px-4 sm:px-6 lg:px-8"
    >
      <div
        class="flex justify-center items-center h-64"
      >
        <div
          class="_skeleton_8f912d"
          style="width: 200px; height: 300px; border-radius: 0px;"
        />
      </div>
    </section>
  </div>
</div>
`;
